js/customize-snapshots-front.js

```diff
@@ -0,0 +1,65 @@
+/* global jQuery, wp */
+/* exported CustomizeSnapshotsFront */
+var CustomizeSnapshotsFront = (function( $ ) {
+	'use strict';
+
+	var component = {
+		data: {
+			confirmationMsg: '',
+			action: '',
+			snapshotsFrontendPublishNonce: '',
+			errorMsg: ''
+		}
+	};
+	/**
+	 * Init.
+	 *
+	 * @param {object} args Args.
+	 * @returns {void}
+	 */
+	component.init = function init( args ) {
+		_.extend( component.data, args );
+
+		$( document ).ready( function() {
+			component.setupPublishButton();
+		} );
+	};
+
+	/**
+	 * Set up snapshot frontend publish button.
+	 *
+	 * @returns {void}
+	 */
+	component.setupPublishButton = function setupPublishButton() {
+		var publishBtn = $( '#wp-admin-bar-publish-customize-snapshot a' );
+
+		if ( ! publishBtn.length ) {
+			return;
+		}
+
+		publishBtn.click( function( e ) {
+			var request;
+			e.preventDefault();
+
+			if ( ! window.confirm( component.data.confirmationMsg ) ) { // eslint-disable-line no-alert
+				return false;
+			}
+			request = wp.ajax.post( component.data.action, {
+				nonce: component.data.snapshotsFrontendPublishNonce,
+				uuid: component.data.uuid
```

Comment:
The only hesitation to including the `theme` in the preview link is that if a user is not authenticated they won't be able to access the preview, since previewing a theme switch requires the `switch_themes` capability.